tests/common/widgets/dynamic-widgets/options.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  createAlgoliaSearchClient,
+  createMultiSearchResponse,
+  createSingleSearchResponse,
+} from '@instantsearch/mocks';
+import { wait } from '@instantsearch/testutils';
+import { within, screen } from '@testing-library/dom';
+
+import type { DynamicWidgetsWidgetSetup } from '.';
+import type { TestOptions } from '../../common';
+
+export function createOptionsTests(
+  setup: DynamicWidgetsWidgetSetup,
+  { act }: Required<TestOptions>
+) {
+  describe('options', () => {
+    const searchClient = createAlgoliaSearchClient({
+      search: jest.fn((requests) => {
+        return Promise.resolve(
+          createMultiSearchResponse(
+            ...requests.map(() => {
+              return createSingleSearchResponse({
+                facets: {
+                  brand: {
+                    Samsung: 633,
+                    Metra: 591,
+                  },
+                  category: {
+                    TV: 633,
+                    Radio: 591,
+                  },
+                },
+                renderingContent: {
+                  facetOrdering: {
+                    facets: {
+                      order: ['category', 'brand'],
+                    },
+                  },
+                },
+              });
+            })
+          )
+        );
+      }),
+    });
+
+    beforeEach(() => {
+      searchClient.search.mockClear();
+    });
```

Comment:
Nit, but in other tests we wrap this into a `createMockedSearchClient` function and instantiate a new client in each test. This would work the same, but the benefits are that you don't need to clear between tests, and you can keep the mock at the end of the file so it's not the first thing you see when reading the tests.